### DISCUSSION TITLE
Khmer (Cambodia) language Tokenizer

### DISCUSSION THREAD

**Discussion by cakhsoft** [meilisearch/product#374] [Feedback & Feature Proposal]
Please help make Khmer language tokenizer support. Thank you!

**Comment by curquiza** [comment]
Hello @smartrides 

Thanks for your request, I move this feature request to the right repository :)

**Comment by ManyTheFish** [comment]
Linking a related discussion below:
https://github.com/meilisearch/product/discussions/316

**Comment by xshadowlegendx** [comment]
there is `icu` library from icu4x in rust that I have tested can segment khmer text but it is in alpha stage now, I think I can try contributing by using the library, do you think that works?

**Reply by ManyTheFish** [reply]
Hello @xshadowlegendx,
Feel free to make a Pull Request on [Charabia's repository](https://github.com/meilisearch/charabia), please read the CONTRIBUTING.md file, you will find some help to implement the segmenter.

About the alpha stage of the library, let's put it behind a feature flag, this way, we will be able to release an early prototype of Meilisearch for the Khmer Language support allowing you to try this enhancement. Moreover, you'll be able to share this prototype with people interested in this feature.

Thanks!

**Reply by xshadowlegendx** [reply]
hello @ManyTheFish ok I will create the pr and working on it
